src/parser/statement.js

```diff
@@ -78,37 +88,57 @@ pp.parseStatement = function (declaration, topLevel) {
       if (!declaration) this.unexpected();
       return this.parseClass(node, true);
 
-    case tt._if: return this.parseIfStatement(node);
-    case tt._return: return this.parseReturnStatement(node);
-    case tt._switch: return this.parseSwitchStatement(node);
-    case tt._throw: return this.parseThrowStatement(node);
-    case tt._try: return this.parseTryStatement(node);
+    case tt._if:
+      return this.parseIfStatement(node);
+    case tt._return:
+      return this.parseReturnStatement(node);
+    case tt._switch:
+      return this.parseSwitchStatement(node);
+    case tt._throw:
+      return this.parseThrowStatement(node);
+    case tt._try:
+      return this.parseTryStatement(node);
 
     case tt._let:
     case tt._const:
-      if (!declaration) this.unexpected(); // NOTE: falls through to _var
-
+      if (!declaration) this.unexpected();
+    // NOTE: falls through to _var
     case tt._var:
       return this.parseVarStatement(node, starttype);
 
-    case tt._while: return this.parseWhileStatement(node);
-    case tt._with: return this.parseWithStatement(node);
-    case tt.braceL: return this.parseBlock();
-    case tt.semi: return this.parseEmptyStatement(node);
+    case tt._while:
+      return this.parseWhileStatement(node);
+    case tt._with:
+      return this.parseWithStatement(node);
+    case tt.braceL:
+      return this.parseBlock();
+    case tt.semi:
+      return this.parseEmptyStatement(node);
     case tt._export:
     case tt._import:
-      if (this.hasPlugin("dynamicImport") && this.lookahead().type === tt.parenL) break;
+      if (
+        this.hasPlugin("dynamicImport") && this.lookahead().type === tt.parenL
```

Comment:
I opened https://github.com/prettier/prettier/issues/868 as it looks a bit weird